docs/nixl.md

```
@@ -177,3 +177,34 @@ Invalidate local agent metadata()
 deregister memory regions # optional
 delete agent
 ```
+
+# Configuration
+
+NIXL and its plugins have config options that can be set via environment variables and/or a config file.
+
+When a config option named `NIXL_OPTION_FOO` is requested first the environment and then the config file is checked.
+If the option exists in the environment the config file is not consulted, even in case of value conversion errors.
+
+### Config Files
```

markdownlint-cli2 (0.22.1)

[warning] Heading levels should only increment by one level at a time
Expected: h2; Actual: h3

(MD001, heading-increment)